src/storage/local.rs

```diff
@@ -0,0 +1,69 @@
+#![allow(missing_docs)]
+
+use std::vec::Vec;
+use std::path::{Path, PathBuf};
+
+use core::LalResult;
+
+
+#[derive(Serialize, Deserialize, Clone, Default)]
+pub struct LocalConfig {}
+
+use super::{Backend, Component};
+
+pub struct LocalBackend {
+    /// Local config
+    pub config: LocalConfig,
+    /// Cache directory
+    pub cache: String,
+}
+
+impl LocalBackend {
+    pub fn new(cfg: &LocalConfig, cache: &str) -> Self {
+        LocalBackend{
+            config: cfg.clone(),
+            cache: cache.into(),
+        }
+    }
+}
+
+/// Artifact backend trait for `LocalBackend`
+///
+/// This is intended to be used by the caching trait `CachedBackend`, but for
+/// specific low-level use cases, these methods can be used directly.
+impl Backend for LocalBackend {
+    fn get_versions(&self, _name: &str, _loc: Option<&str>) -> LalResult<Vec<u32>> {
+    	Ok(vec![])	
+    }
+
+    fn get_latest_version(&self, _name: &str, _loc: Option<&str>) -> LalResult<u32> {
+    	Ok(0)
```

Comment:
so there's only ever a version 0? then the previous vector should be `vec![0]` right?

Reply:
I'm just appeasing the type system. Without somewhere to store artifacts in `LocalBackend`, versions are meaningless.

I'm open to ideas about what we do about locally publishing artifacts in this backend.

Reply:
I feel like if we just made lal publish put things directly in stash rather than upload somewhere, then a `LocalBackend` can just query stash and avoid HTTP altogether. May need to tweak how the traits are set up though.